tensorflow/core/kernels/batch_norm_op_test.cc

```diff
@@ -29,60 +29,58 @@ limitations under the License.
-  TF_EXPECT_OK(InitOpWithGraphVersion(8));
-  AddInputFromArray<float>(TensorShape({1, 1, 6, 2}),
-                           {1, 4, 2, 5, 3, 6, -1, -4, -2, -5, -3, -6});
-  AddInputFromArray<float>(TensorShape({2}), {10, 20});
-  AddInputFromArray<float>(TensorShape({2}), {0.25f, 0.5f});
-  AddInputFromArray<float>(TensorShape({2}), {0.1f, 0.6f});
-  AddInputFromArray<float>(TensorShape({2}), {0.0f, 0.0f});
-  TF_ASSERT_OK(RunOpKernel());
-
-  Tensor expected(allocator(), DT_FLOAT, TensorShape({1, 1, 6, 2}));
-  test::FillValues<float>(
+    TF_EXPECT_OK(InitOpWithGraphVersion(8));
+
+    AddInputFromList<T>(TensorShape({1, 1, 6, 2}),
+                                {1, 4, 2, 5, 3, 6, -1, -4, -2, -5, -3, -6});
+    AddInputFromList<T>(TensorShape({2}), {10, 20});
+    AddInputFromList<T>(TensorShape({2}), {0.25, 0.5});
+    AddInputFromList<T>(TensorShape({2}), {0.1, 0.6});
+    AddInputFromList<T>(TensorShape({2}), {0.0, 0.0});
+
+    TF_ASSERT_OK(RunOpKernel());
+
+    double atol = TValueType == DT_FLOAT ? 0.01 : 0.1;
+
+    Tensor expected(allocator(), TValueType, TensorShape({1, 1, 6, 2}));
+    test::FillValues<T>(
       &expected, {-17.86f, -22.00f, -15.87f, -20.59f, -13.87f, -19.18f, -21.86f,
                   -33.31f, -23.85f, -34.72f, -25.85f, -36.13f});
-  test::ExpectTensorNear<float>(expected, *GetOutput(0), 0.01);
-}
+    test::ExpectTensorNear<T>(expected, *GetOutput(0), atol);
+  }
+};
 
-TEST_F(BatchNormOpTest, Fp16) {
-  TF_EXPECT_OK(
-      NodeDefBuilder("batch_norm_op", "BatchNormWithGlobalNormalization")
-          .Input(FakeInput(DT_HALF))
-          .Input(FakeInput(DT_HALF))
-          .Input(FakeInput(DT_HALF))
-          .Input(FakeInput(DT_HALF))
-          .Input(FakeInput(DT_HALF))
-          .Attr("scale_after_normalization", false)
-          .Attr("variance_epsilon", 0.001)
-          .Finalize(node_def()));
-  TF_EXPECT_OK(InitOpWithGraphVersion(8));
-  AddInputFromList<Eigen::half>(TensorShape({1, 1, 6, 2}),
-                                {1, 4, 2, 5, 3, 6, -1, -4, -2, -5, -3, -6});
-  AddInputFromList<Eigen::half>(TensorShape({2}), {10, 20});
-  AddInputFromList<Eigen::half>(TensorShape({2}), {0.25, 0.5});
-  AddInputFromList<Eigen::half>(TensorShape({2}), {0.1, 0.6});
-  AddInputFromList<Eigen::half>(TensorShape({2}), {0.0, 0.0});
-  TF_ASSERT_OK(RunOpKernel());
-
-  Tensor expected(allocator(), DT_HALF, TensorShape({1, 1, 6, 2}));
-  test::FillValues<Eigen::half>(
-      &expected, {-17.86, -22.00, -15.87, -20.59, -13.87, -19.18, -21.86,
-                  -33.31, -23.85, -34.72, -25.85, -36.13});
-  test::ExpectTensorNear<Eigen::half>(expected, *GetOutput(0), 0.1);
+TYPED_TEST_SUITE_P(BatchNormOpTest);
+
+TYPED_TEST_P(BatchNormOpTest, Simple) {
+  this->run_me();
 }
 
+REGISTER_TYPED_TEST_SUITE_P(BatchNormOpTest,
+                            Simple);
+
+// TODO(ezhulenev): Add support for more data types.
+using DataTypes = ::testing::Types<float, Eigen::half>;//, Eigen::bfloat16>;
```

Comment:
Are you planning to add bfloat16 here?  Why is that commented?

Reply:
I have rechecked it, the point is, there is no registered BatchNormWithGlobalNormalization OpKernel for bfloat16 number type: https://github.com/tensorflow/tensorflow/blob/master/tensorflow/core/kernels/batch_norm_op.cc#L197-L203 
I will remove it then..